compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -427,29 +427,20 @@ public void generateVarLoad(MethodVisitor mv, BIRNode.BIRVariableDcl varDcl, int
         BType bType = JvmCodeGenUtil.getReferredType(varDcl.type);
 
         switch (varDcl.kind) {
-            case GLOBAL: {
-                BIRNode.BIRGlobalVariableDcl globalVar = (BIRNode.BIRGlobalVariableDcl) varDcl;
-                String moduleName = JvmCodeGenUtil.getPackageName(globalVar.pkgId);
-
-                String varName = varDcl.name.value;
-                String className = jvmPackageGen.lookupGlobalVarClassName(moduleName, varName);
-
-                String typeSig = getTypeDesc(bType);
-                mv.visitFieldInsn(GETSTATIC, className, varName, typeSig);
-                return;
-            }
             case SELF:
                 mv.visitVarInsn(ALOAD, 0);
                 return;
-            case CONSTANT: {
+            case CONSTANT:
+            case GLOBAL:
                 String varName = varDcl.name.value;
                 PackageID moduleId = ((BIRNode.BIRGlobalVariableDcl) varDcl).pkgId;
                 String pkgName = JvmCodeGenUtil.getPackageName(moduleId);
                 String className = jvmPackageGen.lookupGlobalVarClassName(pkgName, varName);
                 String typeSig = getTypeDesc(bType);
                 mv.visitFieldInsn(GETSTATIC, className, varName, typeSig);
                 return;
-            }
+            default:
+                break;
```

Comment:
Is the `default` case with `break` needed here?

Reply:
Yes, it is considered as a good coding practice to always add the default case in switch case. Sonarlint suggests that.

Reply:
We are not practicing that in most of the places in our code base (when default case does not seem necessary). We can adopt this in future as a coding best practice.